cpp/mg_utility/mg_graph.hpp

```diff
@@ -0,0 +1,264 @@
+/// @file graph.hpp
+/// This file contains the graph definitions.
+
+#pragma once
+
+#include <algorithm>
+#include <map>
+#include <unordered_map>
+
+#include "data_structures/graph_data.hpp"
+#include "data_structures/graph_view.hpp"
+#include "mg_exceptions.hpp"
+
+namespace mg_graph {
+
+/// Graph representation.
+template <typename TSize> class Graph : public GraphView<TSize> {
+
+  using TNode = Node<TSize>;
+  using TEdge = Edge<TSize>;
+  using TNeighbour = Neighbour<TSize>;
```

Comment:
I think you can do something like
```
using typename Node<TSize>::TNode;
```
It's equally long but I think it's safer so the aliases are in sync.

Reply:
This breaks my code with the following exception:
`error: type ‘mg_graph::Node<long unsigned int>’ is not a base type for type ‘mg_graph::GraphView<long unsigned int>` Are you sure this is the right way?

Reply:
I meant the base class:
```
using typename GraphView<TSize>::TNode;
```
Sorry!